src/cpp/session/modules/SessionCpp.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * SessionCpp.cpp
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include "SessionCpp.hpp"
+
+#include <shared_core/Error.hpp>
+#include <shared_core/FilePath.hpp>
+
+#include <core/Exec.hpp>
+#include <core/json/JsonRpc.hpp>
+#include <core/r_util/RPackageInfo.hpp>
+#include <core/regex/RegexMatch.hpp>
+
+#include <session/SessionModuleContext.hpp>
+#include <session/projects/SessionProjects.hpp>
+
+#include <string>
+
+using namespace rstudio::core;
+
+namespace rstudio {
+namespace session {
+namespace modules {
+namespace cpp {
+
+namespace {
+
+Error cppProjectStyle(const json::JsonRpcRequest& request,
+                          json::JsonRpcResponse* pResponse)
+{
+   projects::ProjectContext& context = projects::projectContext();
+   
+   // ----- not in a project or not in a package: give up
+   if (!context.hasProject() || !context.isPackageProject())
+   {
+      pResponse->setResult("");
+      return Success();
+   }
+
+   // ----- Look for Rcpp or cpp11 in LinkingTo
+   const core::r_util::RPackageInfo& info = context.packageInfo();
+   const std::string& linkingTo = info.linkingTo();
+   
+   if (core::regex_utils::search(linkingTo, boost::regex("cpp11"))) 
```

Comment:
Nit: avoid making temporaries with `boost::regex()`. I don't think it matters here, but at least for other tools (e.g. regex matching) the lifetime of the pattern object needs to persist beyond the regex calls themselves.

Reply:
Thanks.